service/history/api/getworkflowexecutionrawhistory/api.go

```diff
@@ -0,0 +1,287 @@
+	if err != nil {
+		return nil, err
+	}
+
+	req := request.Request
+	execution := req.Execution
+	var pageToken *tokenspb.RawHistoryContinuation
+	var targetVersionHistory *historyspb.VersionHistory
+	if req.NextPageToken == nil {
+		response, err := api.GetOrPollMutableState(
+			ctx,
+			shardContext,
+			&historyservice.GetMutableStateRequest{
+				NamespaceId: ns.ID().String(),
+				Execution:   execution,
+			},
+			workflowConsistencyChecker,
+			eventNotifier,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		targetVersionHistory, err = setRequestDefaultValueAndGetTargetVersionHistory(
+			request,
+			response.GetVersionHistories(),
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		pageToken = api.GeneratePaginationToken(request, response.GetVersionHistories())
+	} else {
+		pageToken, err = api.DeserializeRawHistoryToken(req.NextPageToken)
+		if err != nil {
+			return nil, err
+		}
+		versionHistories := pageToken.GetVersionHistories()
+		if versionHistories == nil {
+			return nil, consts.ErrInvalidVersionHistories
+		}
+		targetVersionHistory, err = setRequestDefaultValueAndGetTargetVersionHistory(
+			request,
+			versionHistories,
+		)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if err := api.ValidatePaginationToken(
+		request,
+		pageToken,
+	); err != nil {
+		return nil, err
+	}
+
+	pageSize := int(req.GetMaximumPageSize())
+	shardID := common.WorkflowIDToHistoryShard(
+		ns.ID().String(),
+		execution.GetWorkflowId(),
+		shardContext.GetConfig().NumberOfShards,
+	)
+	rawHistoryResponse, err := shardContext.GetExecutionManager().ReadRawHistoryBranch(ctx, &persistence.ReadHistoryBranchRequest{
+		BranchToken: targetVersionHistory.GetBranchToken(),
+		// GetWorkflowExecutionRawHistory is inclusive/inclusive.
+		// ReadRawHistoryBranch is inclusive/exclusive.
+		MinEventID:    pageToken.GetStartEventId(),
+		MaxEventID:    pageToken.GetEndEventId() + 1,
+		PageSize:      pageSize,
+		NextPageToken: pageToken.PersistenceToken,
+		ShardID:       shardID,
+	})
+	if err != nil {
+		if _, isNotFound := err.(*serviceerror.NotFound); isNotFound {
+			// when no events can be returned from DB, DB layer will return
+			// EntityNotExistsError, this API shall return empty response
+			return &historyservice.GetWorkflowExecutionRawHistoryResponse{
+				Response: &adminservice.GetWorkflowExecutionRawHistoryResponse{
+					HistoryBatches: []*commonpb.DataBlob{},
+					NextPageToken:  nil, // no further pagination
+					VersionHistory: targetVersionHistory,
+				},
+			}, nil
+		}
+		return nil, err
+	}
+
+	pageToken.PersistenceToken = rawHistoryResponse.NextPageToken
+	size := rawHistoryResponse.Size
+	metricsHandler := interceptor.GetMetricsHandlerFromContext(ctx, shardContext.GetLogger()).WithTags(metrics.OperationTag(metrics.HistoryGetWorkflowExecutionRawHistoryScope))
+	metricsHandler.Histogram(metrics.HistorySize.Name(), metrics.HistorySize.Unit()).Record(
```

Comment:
Why emit this metric

Reply:
There might be historical reason because I saw the old api is emit this metric